src/app/static/src/app/store/baseline/baseline.ts

```diff
@@ -9,15 +9,18 @@ import {localStorageManager} from "../../localStorageManager";
 export interface BaselineState extends ReadyState {
     selectedDataset: Dataset | null
     pjnzError: Error | null
+    pjnzErroredFile: string | null
```

Comment:
Perhaps you tried this and it threw up other problems but it there a need for extra properties on the state? Couldn't you just construct a fake `PjnzResponse` object with the filename and the `fromADR` flag and save if to the existing `pjnz` property with the existing mutation `PJNZUpdated`? And then that would just filter through to the component without needing to make any changes there?

Reply:
I did consider this, but I thought there were probably multiple places in the code where we are doing checks on `if (state.pjnz)` etc in order to check for successful responses, which would then fail and be a pain to be debug. I didn't actually try it, might be worth it to overcome the ADR issue though.

Reply:
No worries, lets leave this as is then as it works fine :+1: